### DISCUSSION TITLE
Q: Nginx反向代理至非443端口，api接口仍访问443端口 || Q: Nginx reverse proxy to non-443 port, the api interface still accesses 443 port

### DISCUSSION THREAD

**Discussion by Raqtpie** [walinejs/waline#2407] [Q&A] [answered]
### Describe the bug | 问题描述

我使用Nginx反向代理给waline进行https加密，绑定到了8366端口，页面访问正常，但是登录/注册等接口fetch to fail，浏览器调试发现访问的接口链接不带端口，控制台打印locaiton.host返回的数据正常，为url+port，但源代码中的serverUrl不带端口

Waline部署方式：Docker-compose

Nginx配置
```
server {
  listen 8366 ssl;
  server_name walinetest.raqtpie.xyz;

  ssl_certificate /etc/nginx/tls/certificate.crt;
  ssl_certificate_key /etc/nginx/tls/private.pem;

  # proxy to 8360
  location / {
    proxy_pass http://127.0.0.1:8360;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header REMOTE-HOST $remote_addr;
    add_header X-Cache $upstream_cache_status;
    # cache
    add_header Cache-Control no-cache;
    expires 12h;
  }

}
```

相关截图
![Snipaste_2024-03-27_14-08-54](https://github.com/walinejs/waline/assets/109587205/a7b127a0-ccad-4542-91b2-fd2f98ce4b89)
![Snipaste_2024-03-27_14-10-05](https://github.com/walinejs/waline/assets/109587205/a759b591-3327-4f41-8b23-e9eea5a103c6)
![Snipaste_2024-03-27_14-12-00](https://github.com/walinejs/waline/assets/109587205/815af647-1747-47d8-a6d8-c47940ef5828)

### Website URL | 问题网站

https://walinetest.raqtpie.xyz:8366

### Where your waline deploy? | 服务部署在哪里？

Self Host

### Where your comment data store? | 数据存储在哪里？

SQLite

---
### Describe the bug | Problem description

I used Nginx reverse proxy to encrypt https for waline, and bound it to port 8366. The page access was normal, but the login/registration and other interfaces fetched to fail. The browser debugging found that the accessed interface link did not have a port, and the console printed locaiton.host The returned data is normal and is url+port, but the serverUrl in the source code does not have a port.

Waline deployment method: Docker-compose

Nginx configuration
```
server {
  listen 8366 ssl;
  server_name walinetest.raqtpie.xyz;

  ssl_certificate /etc/nginx/tls/certificate.crt;
  ssl_certificate_key /etc/nginx/tls/private.pem;

  # proxy to 8360
  location/{
    proxy_pass http://127.0.0.1:8360;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header REMOTE-HOST $remote_addr;
    add_header X-Cache $upstream_cache_status;
    # cache
    add_header Cache-Control no-cache;
    expires 12h;
  }

}
```

Related screenshots
![Snipaste_2024-03-27_14-08-54](https://github.com/walinejs/waline/assets/109587205/a7b127a0-ccad-4542-91b2-fd2f98ce4b89)
![Snipaste_2024-03-27_14-10-05](https://github.com/walinejs/waline/assets/109587205/a759b591-3327-4f41-8b23-e9eea5a103c6)
![Snipaste_2024-03-27_14-12-00](https://github.com/walinejs/waline/assets/109587205/815af647-1747-47d8-a6d8-c47940ef5828)

### Website URL | Problem website

https://walinetest.raqtpie.xyz:8366

### Where your waline deploy? | Where is the service deployed?

Self Host

### Where your comment data store? | Where is the data stored?

SQLite

**Comment by lizheming** [comment, marked as answer]
参考这个 issue https://github.com/walinejs/waline/issues/2390 自己配下 SERVER_URL 吧
